doc/Programming-guide/Your-first-Kaa-application/attach/demo-cpp/KaaDemo.cpp

```diff
@@ -56,6 +57,11 @@ class ConfigurationCollection : public kaa::IConfigurationReceiver {
     {
         // Run the Kaa endpoint.
         kaaClient_->start();
+        // Read default sample period
+        samplePeriod_ = kaaClient_->getConfiguration().samplePeriod;
```

Comment:
I think kaaClient can receive a new configuration before the default sample period is printed. In this case you'll get this:

> Received configuration data. New sample period: 16

> Default sample period: 16